luskan/src/test/java/org/opencadc/luskan/InventoryFunctionConverterTest.java

```diff
@@ -0,0 +1,220 @@
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.luskan;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+import ca.nrc.cadc.tap.TapQuery;
+import ca.nrc.cadc.tap.schema.TapSchema;
+import ca.nrc.cadc.util.Log4jInit;
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.uws.Parameter;
+
+public class InventoryFunctionConverterTest {
+    private static final Logger log = Logger.getLogger(InventoryFunctionConverterTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.luskan", Level.INFO);
+        Log4jInit.setLevel("net.sf.jsqlparser", Level.INFO);
+    }
+
+    private static final TapSchema tapSchema = TestUtil.mockTapSchema();
+
+    @Test
+    public void testSelect() {
+        String query = "select *, num_copies() from inventory.Artifact where id = '{some artifact id}'";
+        String expected = "SELECT inventory.Artifact.id, inventory.Artifact.contentLength, cardinality(inventory.Artifact.siteLocations) FROM inventory.Artifact WHERE id = '{some artifact id}'";
+        doTest(query, expected, "inventory.Artifact.siteLocations");
+    }
+
+    @Test
+    public void testWhere() {
+        String query = "select count(*) from inventory.Artifact where num_copies() = 3";
+        String expected = "select count(*) from inventory.Artifact where cardinality(inventory.Artifact.siteLocations) = 3";
+        doTest(query, expected, "inventory.Artifact.siteLocations");
+    }
+
+    @Test
+    public void testGroupByAs() {
+        String query = "select count(*), num_copies() as num from inventory.Artifact group by num";
+        String expected = "select count(*), cardinality(inventory.Artifact.siteLocations) as num from inventory.Artifact group by num";
+        doTest(query, expected, "inventory.Artifact.siteLocations");
+    }
+
+    @Test
+    public void testAlias() {
+        String query = "select count(*), num_copies() from inventory.Artifact as a";
+        String expected = "select count(*), cardinality(a.siteLocations) from inventory.Artifact as a";
+        doTest(query, expected, "a.siteLocations");
+    }
+
+    @Test
+    public void testMultipleArtifactTables() {
+        String query = "select count(*), num_copies() from inventory.Artifact as a, temp.Artifact as b where a.id = b.id";
+        String expected = "select count(*), cardinality(a.siteLocations) from inventory.Artifact as a, temp.Artifact as b where a.id = b.id";
+        doTest(query, expected, null);
```

Comment:
could have `"a.siteLocations"` as expected cardinality here... I was expecting 
`"cardinality(a.siteLocations)"` as the string to look for in all these cases, just to be as explicit as possible.

Reply:
Missed that, the expected function should be been given, fixed.